### PR TITLE
timeline: update replies when a message has been edited

### PR DESCRIPTION
**Pull request by bnjbvr**
When a message B is a reply to message A, an edit to message A should be reflected in the replied-to part of message B.

Part of #3905.

### CODE REVIEW

File: crates/matrix-sdk-ui/src/timeline/event_handler.rs [hunk 1]
```diff
@@ -511,9 +512,37 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
...
+                let internal_id = item.internal_id.to_owned();
+
+                // Update all events that replied to this message with the edited content.
+                self.items.for_each(|mut entry| {
+                    let Some(event_item) = entry.as_event() else { return };
+                    let Some(message) = event_item.content.as_message() else { return };
+                    let Some(in_reply_to) = message.in_reply_to() else { return };
+                    if replacement.event_id == in_reply_to.event_id {
+                        warn!(
+                            "BNJ find one! new reply: {}",
```
Comment (Hywan) [inline crates/matrix-sdk-ui/src/timeline/event_handler.rs:527]: Maybe rephrase the warning message :-)?

File: crates/matrix-sdk-ui/tests/integration/timeline/edit.rs [hunk 2]
```diff
@@ -444,6 +445,104 @@ async fn test_send_reply_edit() {
...
+    let f = EventFactory::new();
+    let event_id = event_id!("$original_event");
+    let user_id = client.user_id().unwrap();
+
+    // When a room has two messages, one is a reply to the other…
+    sync_builder.add_joined_room(
+        JoinedRoomBuilder::new(room_id)
+            .add_timeline_event(f.text_msg("bonjour").sender(user_id).event_id(event_id))
+            .add_timeline_event(f.text_msg("hi back").reply_to(event_id).sender(*ALICE)),
+    );
```
Comment (Hywan) [inline crates/matrix-sdk-ui/tests/integration/timeline/edit.rs:477]: Can you test with 2 events replying to `event_id` please? The idea is to ensure we are looping over all events that are a reply to the edited event.

**Review by Hywan — APPROVED** [review]
Thanks, it looks good! I've one feedback regarding the test (an important one), and one regarding a log message. I'm approving this PR to unblock you though.

File: crates/matrix-sdk-ui/tests/integration/timeline/edit.rs (hunk 2 above)
Comment (bnjbvr) [reply crates/matrix-sdk-ui/tests/integration/timeline/edit.rs:477]: Sure!
